tests/by-util/test_date.rs

```diff
@@ -395,3 +395,12 @@ fn test_invalid_date_string() {
         .no_stdout()
         .stderr_contains("invalid date");
 }
+
+#[test]
+fn test_issue_5149() {
```

Comment:
Let's name this test something like `test_datetime_overflow` so that we can easily tell what it is testing.